apps/condo/domains/billing/utils/countrySpecificQRCodeParsers/ru.qrCodeParser.js

```diff
@@ -17,21 +17,20 @@ const { get } = require('lodash')
  * @return {TRUQRCodeFields}
  */
 function parseRUReceiptQRCode (qrStr) {
-    const matches = /^ST(?<version>\d{4})(?<encodingTag>\d)\|(?<requisitesStr>.*)$/g.exec(qrStr)
+    const buffer = Buffer.from(qrStr, 'base64')
+    const { encoding: detectedEncoding } = jschardet.detect(buffer)
 
-    if (!matches) {
-        throw new Error('Invalid QR code')
-    }
+    // In Russia, only two formats are used: utf and windows-1251. Therefore, here we strictly control the formats
+    const encoding = detectedEncoding && detectedEncoding.toUpperCase() === 'UTF-8' ? 'UTF-8' : 'WINDOWS-1251'
```

Comment:
In the case of undefined encoding you will get cp1251
<img width="646" alt="image" src="https://github.com/user-attachments/assets/4aad8f52-903e-44f4-8f66-ef2b8e08ba1b">

Reply:
Why do you ignoring other possible encodings?

Reply:
This is country specific logic. This country uses formats that parse well in 1251. Therefore, we use only UTF-8 and win1251

Reply:
> In the case of undefined encoding you will get cp1251 <img alt="image" width="646" src="https://private-user-images.githubusercontent.com/25384290/368031650-4aad8f52-903e-44f4-8f66-ef2b8e08ba1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY1NjYxMDAsIm5iZiI6MTcyNjU2NTgwMCwicGF0aCI6Ii8yNTM4NDI5MC8zNjgwMzE2NTAtNGFhZDhmNTItOTAzZS00NGY0LThmNjYtZWYyYjhlMDhiYTFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE3VDA5MzY0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxYTU2ZmY2Njk5M2RiMDgyOTg5ZGZlYjNiY2MzNmJmN2ZjOGY2MTZjM2Q2MTEwNGY0Yjk2NWYyYTkxYjQ5MzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.imN0DNOBPf0Ccog3uXdwMNMOUnbgN09YMw2yzHzknTc">

if the format is not parsed, then the data will not be visible by the regular expression. So it's invalid and throw Error
<img width="894" alt="image" src="https://github.com/user-attachments/assets/383182f8-5be9-4bf3-8acc-01a770feed8e">